consensus/src/processes/mass.rs

```diff
@@ -91,10 +91,13 @@ impl MassCalculator {
         let ins_len = tx.tx().inputs.len() as u64;
 
         /*
-           KIP-0009 relaxed formula for the cases |O| = 1 OR |O| <= |I| <= 2:
-               max( 0 , C·( |O|/H(O) - |I|/H(I) ) )
+          KIP-0009 relaxed formula for the cases |O| = 1 OR |O| <= |I| <= 2:
+              max( 0 , C·( |O|/H(O) - |I|/H(I) ) )
+
+           Note: in the case |I| = 1 both formulas are equal, yet the following code (harmonic_ins) is a bit more efficient.
+                 Hence, we transform the condition to |O| = 1 OR |I| = 1 OR |O| = |I| = 2 which is equivalent (and faster).
         */
-        if version == Kip9Version::Beta && (outs_len == 1 || (outs_len <= ins_len && ins_len <= 2)) {
+        if version == Kip9Version::Beta && (outs_len == 1 || ins_len == 1 || (outs_len == 2 && ins_len == 2)) {
```

Comment:
```suggestion
        if version == Kip9Version::Beta && matches!((outs_len, ins_len), (1, _) | (_, 1) | (2, 2)) {
```
not sure about that

Reply:
It's a cool comment that's for sure. I'll check the compiler output for both

Reply:
I have this pattern committed to memory, thanks @biryukovmaxim